OpenRA.Game/Traits/TraitsInterfaces.cs

```diff
@@ -136,6 +136,12 @@ public interface ITooltipInfo
 		bool IsOwnerRowVisible { get; }
 	}
 
+	public interface IProvideTooltipInfo
+	{
+		bool IsVisible(Player forPlayer);
+		string Text { get; }
```

Comment:
Since this is an interface and the names are pretty generic, could you rename those two fields to `IsTooltipVisible` and `TooltipText`, for example?  Just to avoid potential collisions with other interfaces.